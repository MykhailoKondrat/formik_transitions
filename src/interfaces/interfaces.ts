import { FieldInputProps, FieldMetaProps } from "formik";

interface InputMeta {
  field?: FieldInputProps<string | undefined>;
  meta?: FieldMetaProps<string | undefined>;
}
export interface TwoOrMoreArray<T> extends Array<T> {
  0: T;
  1: T;
}

export interface MinimumRequiredInputProps {
  name: string;
  label: string;
}
export interface MyInput extends InputMeta, MinimumRequiredInputProps {
  id?: string;
  placeholder: string;
}
export interface IRadioGroupInput extends InputMeta, MinimumRequiredInputProps {
  options: TwoOrMoreArray<string>;
}
export interface ICheckboxes extends InputMeta, MinimumRequiredInputProps {
  options: Array<string>;
}
export interface MySelect {
  id?: string;
  name: string;
  label: string;
  children?: React.ReactNode;
}

export interface InitialValues {
  firstName: string | undefined;
  workType: string | undefined;
  checked: Array<string>;
}
