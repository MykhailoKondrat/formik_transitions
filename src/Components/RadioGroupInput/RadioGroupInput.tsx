import {
  FormControl,
  FormControlLabel,
  FormHelperText,
  FormLabel,
  Grid,
  Radio,
  RadioGroup,
} from "@material-ui/core";

import React from "react";
import { IRadioGroupInput } from "../../interfaces/interfaces";
const RadioGroupInput: React.FC<IRadioGroupInput> = ({
  label,
  name,
  options,
  field,
  meta,
  ...props
}: IRadioGroupInput & { children?: React.ReactNode }) => {
  return (
    <FormControl error={meta?.touched && meta.error ? true : undefined}>
      <FormLabel component="legend">{label}</FormLabel>
      <RadioGroup name={name} {...field}>
        {options.map((radioItem) => {
          return (
            <FormControlLabel
              key={radioItem}
              value={radioItem}
              control={<Radio />}
              label={radioItem}
            />
          );
        })}
      </RadioGroup>
      <FormHelperText>{meta?.touched ? meta?.error : undefined}</FormHelperText>
    </FormControl>
  );
};

export default RadioGroupInput;
export {};
