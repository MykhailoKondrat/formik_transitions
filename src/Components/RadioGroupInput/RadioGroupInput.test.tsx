import RadioGroupInput from "./RadioGroupInput";
import React from "react";
import * as Adapter from "../../../setupTests";
import { mount, shallow } from "enzyme";
import { IRadioGroupInput, TwoOrMoreArray } from "../../interfaces/interfaces";
import { FormControlLabel } from "@material-ui/core";
const adapter = Adapter;

// describe("<RadioGroupInput/>", () => {
//   const testOptions: TwoOrMoreArray<string> = ["1", "2", "3"];
//   const expectedResult = 3;
//   const mockProps: IRadioGroupInput = {
//     label: "label",
//     name: "radioInput",
//     options: testOptions,
//   };
//   const wrapper = shallow(<RadioGroupInput {...mockProps} />);
//   it("should match Snapshot", function () {
//     expect(wrapper).toMatchSnapshot();
//   });
//   it("should render correct amout of childs based on passed ", () => {
//     expect(wrapper.find(FormControlLabel).length).toBe(expectedResult);
//   });
// });

const testOptions: TwoOrMoreArray<string> = ["1", "2", "3"];
const expectedResult = 3;
const mockProps: IRadioGroupInput = {
  label: "label",
  name: "radioInput",
  options: testOptions,
};
const wrapper = shallow(<RadioGroupInput {...mockProps} />);

test("should match Snapshot", function () {
  expect(wrapper).toMatchSnapshot();
});
test("should render correct amout of childs based on passed ", () => {
  expect(wrapper.find(FormControlLabel).length).toBe(expectedResult);
});
