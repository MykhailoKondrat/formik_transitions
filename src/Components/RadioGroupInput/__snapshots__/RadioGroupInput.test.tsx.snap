// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(FormLabel))
    component="legend"
  >
    label
  </WithStyles(ForwardRef(FormLabel))>
  <ForwardRef(RadioGroup)
    name="radioInput"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={<ForwardRef(WithStyles) />}
      key="1"
      label="1"
      value="1"
    />
    <WithStyles(ForwardRef(FormControlLabel))
      control={<ForwardRef(WithStyles) />}
      key="2"
      label="2"
      value="2"
    />
    <WithStyles(ForwardRef(FormControlLabel))
      control={<ForwardRef(WithStyles) />}
      key="3"
      label="3"
      value="3"
    />
  </ForwardRef(RadioGroup)>
  <WithStyles(ForwardRef(FormHelperText)) />
</WithStyles(ForwardRef(FormControl))>
`;
